main.ts

```diff
@@ -699,21 +699,23 @@ class Translator {
       case ts.SyntaxKind.Parameter:
         var paramDecl = <ts.ParameterDeclaration>node;
         if (paramDecl.dotDotDotToken) this.reportError(node, 'rest parameters are unsupported');
-        if (paramDecl.initializer) this.emit('[');
-        this.visitEachIfPresent(paramDecl.decorators);
-        if (paramDecl.type) {
-          if (paramDecl.name.kind !== ts.SyntaxKind.ObjectBindingPattern) {
-            this.visit(paramDecl.type);
-          } else {
+        if (paramDecl.name.kind === ts.SyntaxKind.ObjectBindingPattern) {
```

Comment:
This is sort of beyond the threshold of a single case block, isn't it? Maybe extract to a separate function?